stl/inc/ranges

```diff
@@ -1140,16 +1140,18 @@ namespace ranges {
         // clang-format on
 
         _NODISCARD friend constexpr _Ioterator operator+(_Ioterator _It, const difference_type _Off) noexcept(
-            noexcept(static_cast<_Wi>(_It._Current + _Off))) /* strengthened */ requires _Advanceable<_Wi> {
-            return _Ioterator{static_cast<_Wi>(_It._Current + _Off)};
+            noexcept(_It._Current += _Off)) /* strengthened */ requires _Advanceable<_Wi> {
+            _It._Current += _Off;
+            return _It;

```

Comment:
There are test failures of the form:

```
D:\build\out\inc\ranges(1143): warning C4242: '+=': conversion from 'const __int64' to '_Wi', possible loss of data
        with
        [
            _Wi=int
        ]
D:\build\out\inc\ranges(1144): warning C4242: '+=': conversion from 'const __int64' to '_Wi', possible loss of data
        with
        [
            _Wi=int
        ]
```

It appears that the `static_cast<_Wi>` previously suppressed them.

Also affects `operator-`.